app/adaptive_tessellation/Collapse.h

```diff
@@ -12,23 +12,37 @@
 #include <wmtk/utils/TupleUtils.hpp>
 #include "AdaptiveTessellation.h"
 #include "wmtk/ExecutionScheduler.hpp"
-using namespace adaptive_tessellation;
-using namespace wmtk;
 
+namespace adaptive_tessellation {
 class AdaptiveTessellationCollapseEdgeOperation : public wmtk::TriMeshOperationShim<
                                                       AdaptiveTessellation,
                                                       AdaptiveTessellationCollapseEdgeOperation,
                                                       wmtk::TriMeshEdgeCollapseOperation>
 {
 public:
-    TriMesh::Tuple return_edge_tuple;
+    struct SeamData
+    {
+        Tuple mirror_edge_tuple;
+        size_t curve_id;
+    };
     struct OpCache
     {
-        size_t v1;
-        size_t v2;
+        TriMesh::Tuple return_edge_tuple;
+        size_t v1; // first vertex index of the edge being collapsed
+        size_t v2; // second vertex index of hte edge being collapsed
+
         double length3d;
+
+        size_t v_top; // remaining vertex of the input triangle being collapsed
+        size_t v_bot; // remaining vertex of the other triangle being collapsed
+
+        // pair of vids that are an existing edge -> mirror edge, curveid
+        std::map<std::array<size_t, 2>, SeamData> mirrored_edges;
```

Comment:
You may want to use an `std::unordered_map<>` rather than a `std::map` here. You can take a look at the `OrderedPairHash` object in `lagrange/utils/hash.h` for a hash operator that works on pairs.